docker/docker/src/main/java/org/arquillian/cube/docker/impl/client/config/CubeContainer.java

```diff
@@ -406,6 +411,31 @@ public void setContainerName(String containerName) {
         this.containerName = containerName;
     }
 
+    public Collection<String> getDependsOn() {
+        return dependsOn;
+    }
+
+    public void setDependsOn(Collection<String> dependsOn) {
+        this.dependsOn = dependsOn;
+    }
+
+    public Collection<String> getDependingContainers() {
+        // Depends on has more priority than links
+        if (dependsOn != null && dependsOn.size() > 0) {
+            return Collections.unmodifiableCollection(dependsOn);
+        } else {
+            if (links != null && links.size() > 0) {
```

Comment:
How about extracting this whole `else` block to a dedicated method with a name explaining what it's supposed to do?